### ISSUE TITLE
Importing of a .jpg fails and makes a "call to sipi failed" - Sipi has to be killed

### ISSUE THREAD

**Issue by mrivoal** [dhlab-basel/Sipi#115]
We updated Knora, Sipi and we switched to last version of GraphDB 8.

But even before that, we noticed that the import of the attached file makes Sipi crash. 

I first tried to import a .jpg and it was fine. Then I tried the one attached and after some time, I got a "call to Sipi failed". Then I tried to import again the previous one, which was successfullu imported before. But after some time I got a new "call to Sipi" failed.

So we had to kill Sipi in order to have it work again.
![sam7962](https://cloud.githubusercontent.com/assets/17294309/22554784/b150bc22-e961-11e6-84d7-dcf054257657.jpg)

(By the way, I am not able to assign nor tag an issue in Sipi repository.)

**Comment by lrosenth** [comment]
Hi marion,

Thanks for the error report. I'll try out whenever I have a good internet connection. Right now I'm on the train somewhere in Utah ;-)

Best Regards,
Lukas

Sent from my iPad

On 2 Feb 2017, at 08:13, Benjamin Geer <notifications@github.com<mailto:notifications@github.com>> wrote:


Assigned #115<https://github.com/dhlab-basel/Sipi/issues/115> to @lrosenth<https://github.com/lrosenth>.

**Comment by mrivoal** [comment]
All right, enjoy ;-)

**Comment by tobiasschweizer** [comment]
I downloaded the file locally as `test_lausanne.jpg` and ran `local/bin/sipi --format=jpx --file=test_lausanne.jpg test.jpx`

It does not terminate and still keeps running. Another jpg file worked fine.

Maybe this is interesting for Lukas:

```
$ file test_lausanne.jpg: 
test_lausanne.jpg: JPEG image data, JFIF standard 1.01, resolution (DPI), density 72x72, segment length 16, Exif Standard: [TIFF image data, big-endian, direntries=9, manufacturer=SAMSUNG, model=SAMSUNG WB750, orientation=upper-left, xresolution=144, yresolution=152, resolutionunit=2, software=111103 , datetime=2014:06:06 09:44:38], baseline, precision 8, 3072x4096, frames 3
```

**Comment by benjamingeer** [comment]
I think you have to test uploading it to the Sipi server using Knora. Perhaps the problem occurs when Sipi tries to create the thumbnail.

**Comment by tobiasschweizer** [comment]
But the file is not written, so it does not work in standalone mode either.

**Comment by lrosenth** [comment]
I’m on it – it’s a read problem of this particulart JPEG. I assume somethiong with metadata..

Lukas

Am 02.02.2017 um 07:32 schrieb Tobias Schweizer <notifications@github.com<mailto:notifications@github.com>>:


But the file is not written, so it does not work in standalone mode either.

**Comment by benjamingeer** [comment]
> But the file is not written, so it does not work in standalone mode either.

I misunderstood, I thought "It does not terminate and still keeps running" meant "It works". :)

**Comment by lrosenth** [comment]
Nope – it’s an inifinite loop reading the XMP header. This file contains a very long XMP header which does not fit into one marker.

I’m working on it ;-)

Lukas
Am 02.02.2017 um 07:36 schrieb Benjamin Geer <notifications@github.com<mailto:notifications@github.com>>:


But the file is not written, so it does not work in standalone mode either.

I misunderstood, I thought "It does not terminate and still keeps running" meant "It works". :)

**Comment by lrosenth** [comment]
Ok, I fixed it. The problem was in reading JPEG-files with large XMP-headers which do not fit into 65535 bytes. There the XMP has to be splitter using several markers (which becomes very complicated). I now skip the XMP-information in such cases…

Fixed in develop (and pushed from the moving train in the Utah desert ;-)

Lukas

Am 02.02.2017 um 07:36 schrieb Benjamin Geer <notifications@github.com<mailto:notifications@github.com>>:


But the file is not written, so it does not work in standalone mode either.

I misunderstood, I thought "It does not terminate and still keeps running" meant "It works". :)
